### ISSUE TITLE
no content in emergency feed

### ISSUE THREAD

**Issue by matthewlswanson** [concerto/concerto#1302]
I am running 2.3.1 on the latest Ubuntu. I was trying to update the plugins with no success using the update button and also via the ruby gems in the Concerto UI. So in the linux cli I ran "sudo gem install" and this updated all the gems. I've tried running this without sudo before but it doesn't work.

I've tried changing the permissions and ownerof /usr/lib/ruby/1.9.1 and opening it all the way up to no avail
Here is the error log I get:

```
Permission denied - /var/lib/gems/1.9.1/specifications/activerecord-4.2.5.gemspec (Errno::EACCES)
  /usr/lib/ruby/1.9.1/rubygems/specification.rb:541:in `read'
  /usr/lib/ruby/1.9.1/rubygems/specification.rb:541:in `load'
  /usr/lib/ruby/1.9.1/rubygems/specification.rb:269:in `block (2 levels) in _all'
  /usr/lib/ruby/1.9.1/rubygems/specification.rb:268:in `each'
 ... redacted ....
```

**Comment by mfrederickson** [comment]
Matt, it looks like you've got rails 4.2 going on, but we have an issue and a branch to go there so I don't think it should be running under 4.2 yet.  That doesn't address the permissions issue but it might lead to other problems as it hasn't been tested yet.

**Comment by mfrederickson** [comment]
Regarding your permissions issue, I've read that you're not supposed to typically install the gems with sudo.  Can you tell me about your setup?  What user what concerto installed under/who owns those files? Did you use the apt-get package to install version 2.3.1 or did you install it another way?  One of those plugins is going to require 2.3.2, so I'm wondering why you don't just update to it?

In the future, if you want to install your gems manually, I would think you'd want to go into the _/usr/share/concerto_ directory and run something like `RAILS_ENV=production bundle install`.

**Comment by matthewlswanson** [comment]
It tells me not to install with sudo but if I don't, it won't work at all. The user that owns concerto is www-data, which is not a user we can run commands as. We started with version 2.0 with apt-get install concerto-lite and have upgraded it twice with apt-get since then. I'll revert to a previous snapshot and try the upgrade to 2.3.2. Should we consider making a concerto user and reinstalling? Not sure what the data migration process is like.

**Comment by matthewlswanson** [comment]
I did the upgrade to 2.3.2 just now in test. Definitely broke the emergency plugin. It still appears in the plugin list, but says not installed and there is no item under settings to configure it

**Comment by mfrederickson** [comment]
I don't think it matters about creating a concerto user.  The vm has one, but all the files are owned by www-data:www-data as you would expect.  On my capistrano deploys of concerto I have a _deploy_ user and all the files are owned by that user.

**Comment by mfrederickson** [comment]
Can you edit that plugin and set it back to _enabled_?

**Comment by mfrederickson** [comment]
Can you also show me the contents of your _/usr/share/concerto/Gemfile-plugins_ file?

**Comment by matthewlswanson** [comment]
It was already set to enabled. Here is what I did to fix it:
1. sudo gem install concerto_emergency
2. cd /var/lib/gems/1.9.1/
3. sudo find -type f -exec chmod o+r {} +
4. sudo find -type d -exec chmod 755 {} +
5. sudo chmod o+x /var/lib/gems/1.9.1/gems/passenger-5.0.4/buildout/support-binaries/PassengerAgent
6. sudo shutdown -r now
7. On Concerto UI: Plugins -> Concerto Emergency -> Edit Plugin -> changed gem version to 0.8
8. Sudo shutdown -r now

Everything works now, only concern is it shows the gem version as locked

Here are the contents of that file:

gem "concerto_weather"

gem "concerto_remote_video"

gem "concerto_simple_rss"

gem "concerto_iframe"

gem "concerto_calendar"

gem "concerto_hardware"

gem "concerto_frontend"

gem "concerto_emergency", "0.8"



Thanks for your quick responses!

**Comment by matthewlswanson** [comment]
I guess I prematurely thought everything was working. When I try to add a screen, I get a 500 error on the frontend: 
 Something went wrong. Rest assured, you have been assigned 0% of the blame.
Sometimes a strategic retreat is the best option.


The error is at the bottom of these files /var/log/apache2/error.log && /usr/share/concerto/log/production.log:
[production.txt](https://github.com/concerto/concerto/files/67065/production.txt)


[error.txt](https://github.com/concerto/concerto/files/67059/error.txt)

**Comment by mfrederickson** [comment]
Can you fix the permissions on your log file? 
```
Rails Error: Unable to access log file. Please ensure that /usr/share/concerto/log/production.log exists and is chmod 0666. 
```

**Comment by mfrederickson** [comment]
It cannot find your emergency feed.  I need to make it handle that, but in the meantime... you might need to create your emergency feed... just dont have any content

**Comment by matthewlswanson** [comment]
Viola! I don't know why that always gets changed. Everything is almost working now. I changed the feed to "Emergency" and the template to "Simplicity".

When I add content to that feed, the screen goes black but no content shows up.

**Comment by mfrederickson** [comment]
What do you mean by "always gets changed"?
They simplicity feed only shows graphics-- what content did you add to your emergency feed?

**Comment by matthewlswanson** [comment]
It seems like every time I upgrade Concerto, I get that error and have to change it back, that's all. And I am just using a simple .jpg image to test

**Comment by matthewlswanson** [comment]
I've also tried BlueSwoosh. The template loads but the image in the graphics section does not

**Comment by mfrederickson** [comment]
While working with Matthew on this we were able to ascertain via the rails console that it was retrieving the graphic submission for the graphic field just as it should https://github.com/concerto-addons/concerto-emergency/blob/master/lib/concerto_emergency/engine.rb#L70 but when running http://hisconcertourl/frontend/51/fields/1/contents.json he was getting back an empty array.

The Gemfile.lock and apache error log and production log all looked fine and didn't provide any helpful info.

He did mention that he is using Chrome, and the Screen | Preview worked fine for showing the emergency feed information on his mac.

**Comment by mfrederickson** [comment]
I changed the title (to reflect the current issue you are having) and removed a fair amount of the log file from the initial post (for ease of reading).

**Comment by augustf** [comment]
@mfrederickson Is this still an issue we should be trying to reproduce?

**Comment by mfrederickson** [comment]
I'd like to think that this will be fixed when #1329 is fixed.  Perhaps we could retest it then?  What do you think @matthewlswanson is this still an issue for you?

**Comment by matthewlswanson** [comment]
Sounds good to me, and I'd be willing to test it on my end.
